flink/src/main/java/org/apache/iceberg/flink/IcebergStreamWriter.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink;
+
+import java.io.IOException;
+import org.apache.flink.streaming.api.operators.AbstractStreamOperator;
+import org.apache.flink.streaming.api.operators.BoundedOneInput;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.io.TaskWriter;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+
+class IcebergStreamWriter<T> extends AbstractStreamOperator<DataFile>
+    implements OneInputStreamOperator<T, DataFile>, BoundedOneInput {
+
+  private static final long serialVersionUID = 1L;
+
+  private final String tablePath;
+
+  private transient TaskWriterFactory<T> taskWriterFactory;
+  private transient TaskWriter<T> writer;
+  private transient int subTaskId;
+  private transient int attemptId;
+
+  IcebergStreamWriter(String tablePath, TaskWriterFactory<T> taskWriterFactory) {
```

Comment:
Minor: I recommend using `table.name()` instead of `table.location()`. The name is set by the catalog, so it should be how the table was identified. For Hive, it is `catalog.db.table` and for Hadoop it is the location. So it is a more natural identifier to use.

Reply:
You mean `Table.toString()`?

Reply:
Yeah. We can also add `name` to the table interface. I didn't realize we were still using `toString`.